### PR TITLE
Update to Python 3.10 and bump necessary dependencies

### PR DESCRIPTION
**Pull request by smcclure17**
From [SOW 3](https://docs.google.com/document/d/1WLW3EkmjjS1abJeZGvopjUXNGfMKOO_iLxqeB7fyCuw/edit)

This PR

- Updates us to Python 3.10 (and updates dependencies and code to match)
- Pins the Python versions to 3.10 in our GitHub Action workflows
- Removes the deprecated data-monitoring dash app

Snapshot built off these changes:
- [compare](https://covidactnow-git-bump-snapshot-4861-3608-act-now-coalition.vercel.app/internal/compare/)
- [preview](https://covidactnow-git-bump-snapshot-4861-3608-act-now-coalition.vercel.app/?s=48613608)

### CODE REVIEW

File: libs/datasets/nytimes_anomalies.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def date_range_for_row(row: pd.Series):
         ),
         omit_from_rolling_average="yes",
     ).explode(CommonFields.FIPS)
-    df = df.append(df_subgeography_rows)
```
Comment (smcclure17) [inline libs/datasets/nytimes_anomalies.py:114]: `pandas.DataFrame,append()` was deprecated in Pandas 1.4 (we upgraded from 1.2 --> 1.5.3) and should be replaced by `pandas.concat()`. See https://github.com/pandas-dev/pandas/issues/35407

File: libs/datasets/sources/can_scraper_helpers.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def check_variable_coverage(self, variables: List[ScraperVariable]):
         provider_mask = self.all_df[Fields.PROVIDER] == provider_name
         counts = self.all_df.loc[provider_mask, Fields.VARIABLE_NAME].value_counts()
         variables_by_name = {var.variable_name: var for var in variables}
-        for variable_name, count in counts.iteritems():
+        for variable_name, count in counts.items():
```
Comment (smcclure17) [inline libs/datasets/sources/can_scraper_helpers.py:265]: [`pandas.DataFrame.iteritems()`](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.DataFrame.iteritems.html#pandas.DataFrame.iteritems) is deprecated in favor or `...items()`

File: libs/datasets/timeseries.py [hunk 3]
```diff
@@ -28,7 +28,6 @@
 from datapublic.common_fields import DemographicBucket
 from datapublic.common_fields import FieldName
 from datapublic.common_fields import PdFields
-from pandas.core.generic import FrameOrSeries
```
Comment (smcclure17) [inline libs/datasets/timeseries.py:31]: `FrameOrSeries` is no longer accessible, so i've just used `Union[pd.DataFrame, pd.Series]` instead

File: .github/workflows/deploy_api.yml [hunk 4]
```diff
@@ -95,13 +95,6 @@ jobs:
           ${{ runner.os }}-pip-
           ${{ runner.os }}-
 
-    # If you don't manually install the correct version of numpy before
-    # installing modules from requirements.txt, fastparquet may try to install a
-    # newer version of numpy that requires python 3.8 (but we use 3.7).
-    # See https://github.com/dask/fastparquet/issues/632#issuecomment-877871388
-    - run: pip install numpy==1.21.4
-      working-directory: ./covid-data-model
```
Comment (smcclure17) [inline .github/workflows/deploy_api.yml:103]: This is resolved by updating to Python 3.10

File: datapublic/common_init.py [hunk 5]
```diff
@@ -36,12 +35,6 @@ def configure_logging(command: Optional[str] = None):
         wrapper_class=structlog.stdlib.BoundLogger,
         processors=[
             structlog.stdlib.add_log_level,  # required before SentryProcessor()
-            # sentry_sdk creates events for level >= ERROR. Getting breadcrumbs from structlog isn't supported
-            # without a lot of custom work. See https://github.com/kiwicom/structlog-sentry/issues/25.
-            # The SentryJsonProcessor is used to protect against event duplication.
-            # It adds loggers to a sentry_sdk ignore list, making sure that the message logged is
-            # not reported in addition to the exception stack trace.
-            SentryJsonProcessor(level=logging.ERROR, tag_keys="__all__"),
```
Comment (smcclure17) [inline datapublic/common_init.py:44]: `SentryJsonProcessor` was [removed in version 2.0.0](https://github.com/kiwicom/structlog-sentry/releases/tag/v2.0.0). IIRC, we don't track sentry alerts anymore, so I've opted just to remove these configuration options. 

If we want to preserve this, I can either pin the versions to `1.x` or work to refactor this in version `2.0`

File: libs/datasets/statistical_areas.py [hunk 6]
```diff
@@ -71,7 +71,6 @@ def from_local_public_data() -> "CountyToCBSAAggregator":
         df = pd.read_excel(
             dataset_utils.REPO_ROOT / CBSA_LIST_PATH,
             header=2,
-            convert_float=False,
```
Comment (smcclure17) [inline libs/datasets/statistical_areas.py:74]: [`convert_float`](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.read_excel.html) is deprecated. See https://github.com/pandas-dev/pandas/issues/41127 for some more context

File: libs/datasets/sources/hsa_population.py [hunk 7]
```diff
@@ -22,9 +22,9 @@ def get_location_level(location_id):
 
 
 class HSAPopulation(data_source.DataSource):
-    """HSA number and HSA population for each US county.
```
Comment (smcclure17) [inline libs/datasets/sources/hsa_population.py:25]: Black formatting made these changes

File: data/multiregion.json [hunk 8]
```diff
@@ -2,9 +2,9 @@
   "dataset_type": "multiregion",
   "path": "data/multiregion.csv",
   "model_git_info": {
-    "sha": "af47998f569aa7763b494629197f6e474b823fa5",
-    "branch": "main",
+    "sha": "b4f4e571a2bb8857de3d46a91b55cab8d5f4705f",
+    "branch": "smcclure17/deps",
```
[the remark this answers is not in the document]
Comment (smcclure17) [reply data/multiregion.json:6]: Ah, yes--these are an artifact of testing/running the pipeline off this branch--will revert before merging!

**Comment by smcclure17** [comment]
Found a snag with the AWS Infrastructure deploy [(see pipeline run)](https://github.com/act-now-coalition/covid-data-model/actions/runs/7846457172/job/21413196399):

It seems to be importing the wrong pytest version. Pytest `6.2.5` is required for Python 3.10 (which we upgraded to in this PR), but the action still imports `6.2.1`. It then throws an error that appears to be the same as https://github.com/pytest-dev/pytest/discussions/9195

I'll investigate tonight 👀. I think (hope) it might just be a caching issue.

These are the versions the action runs with (notice that pytest is the wrong version):
```
platform linux -- Python 3.10.13, pytest-6.2.1, py-1.11.0, pluggy-0.13.1
```

**Comment by smcclure17** [comment]
@BrettBoval I've run into a slight hiccup with the AWS API infra. deploy--

With the requirements updates, the function package exceeded the maximum bundle size allowed for CloudFront/Lambda functions. I've done some work to trim the dependencies used by the API (e.g., by removing Sentry logging), but I am now getting 503 errors on the dev tier. I need to do some work to identify the exact issue (the tests run fine locally). 

For debugging, I'd like to jump into the Cloudwatch logs, but I've lost our AWS (12-digit) organization ID. Is there any chance you have and could share this code?

TL:DR: The requirement updates pushed us over the maximum lambda size for the API functions (e.g. registering an API key) and I need to do a bit more work to get us successfully down below the size limit 🤕

**Comment by smcclure17** [comment]
@BrettBoval I've split the AWS/API migration into a new PR at #1393 for clarity. I think we should be good to go on this PR. 

Would you be able to put eyes on #1393  at some point? No rush, as these are separate/standalone services that can be deployed or upgraded separately.
